README.md

```diff
@@ -1,74 +1,96 @@
 [![PyPI version](https://badge.fury.io/py/dsp-tools.svg)](https://badge.fury.io/py/dsp-tools)
 
 # DSP-TOOLS - DaSCH Service Platform Tools
-
-dsp-tools is a command line tool that helps you interact with the DaSCH Service Platform API.
-
+dsp-tools is a command line tool that helps you interacting with the DaSCH Service Platform API.  
 Go to [Full Documentation](https://dasch-swiss.github.io/dsp-tools/)
 
 ## Information for developers
+There is a `Makefile` for all the following tasks (and more). Type `make` to print the available targets. 
+
+For a quick start, use: 
+```
+pip install pipenv
+pipenv install --dev
+make run
+```
 
-There is a `Makefile` for all the following tasks (and more). Type `make` to print the available targets. For a quick
-start use:
+This creates a pipenv-environment, installs all dependencies, and installs `dsp-tools` from source.
+
+If you prefer getting around pipenv, use instead:
 ```bash
 make upgrade-dist-tools
 make install-requirements
 make run
 ```
 
-## Install from source
+## Pipenv
+We use pipenv for our dependency management. There are two ways to get started:
+ - `pipenv install --dev` installs all dependencies, while giving them the opportunity to update themselves
+ - `pipenv install --ignore-pipfile` is used to get a deterministic build in production
 
-To install from source run:
+This works because there are two files defining the dependencies:
+ - `Pipfile` replaces `requirements.txt`, but lists only the core dependencies, ordered in two sections:
+   - `[packages]` lists the dependencies used to run the software.
+   - `[dev-packages]` lists additional dependencies used for tests and deployment.
+ - `Pipfile.lock` enables deterministic builds, by exactly pinning the version of all (sub-) dependencies. 
+   This is done automatically, you must not edit `Pipfile.lock`.
 
-```bash
-python3 setup.py install
-```
+The diverse `requirements.txt` files in this repo are only present for backwards compatibility
+and for Github CI.
 
-## Install the requirements
+If you want to install a new package, install it with `pipenv install package`. This 
+ - installs the package (incl. sub-dependencies) in your virtual environment.
+ - adds the package to the section `[packages]` of `Pipfile`. By default, no versions are pinned.
+ - adds the pinned versions of package and all sub-dependencies to `Pipfile.lock`.
 
-To install the requirements run:
+If a package is only needed for development, please install it with `pipenv install package --dev`,
+so it gets added to the `[dev-packages]` section of `Pipfile`.
 
-```bash
-pip3 install -r requirements.txt
-```
+For security reasons, the maintainer regularly executes
+ - `pipenv check` to get informed about vulnerabilities. 
+ - `pipenv lock` to update `Pipfile.lock` with the latest version of every package
```

Comment:
I'm a bit confused if this is correct. Does `pipenv lock` really search for updates of the packages? For me it would seem more logical to execute `pipenv update`, but according to the docs, `pipenv update` does something else, namely:
pipenv update: runs lock, then sync
pipenv lock: Generates Pipfile.lock.
pipenv sync: Installs all packages specified in Pipfile.lock.

Reply:
You could (or maybe even should) change `lock` to `update` here, so it also executes `sync`.

But the commands make sense, and yes: `lock` rebuilds the lock file according to the pipfile, and since we use `*` for all versions, it will grab the latest version and build the lock file accordingly, i.e. with the latest version.  
All that `sync` does, is updating the virtual environment according to the lock file.  
And then `updates` fist updates the lock file according to the pipfile and then installs this updated lockfile into your virtual environment.

Does that clarify it for you?